src/HDF5Msg.chpl

```diff
@@ -1510,6 +1881,98 @@ module HDF5Msg {
+            )
+        );
+        
+        var subdoms: [fD] domain(1);
+        var segSubdoms: [fD] domain(1);
+        var skips = new set(string);
+        var len: int;
+        var nSeg: int;
+
+        (segSubdoms, nSeg, skips) = get_subdoms(filenames, dset + "/" + SEGMENTED_OFFSET_NAME, validFiles);
+        (subdoms, len, skips) = get_subdoms(filenames, dset + "/" + SEGMENTED_VALUE_NAME, validFiles);
+
+        var segEntry = new shared SymEntry(nSeg, int);
+        read_files_into_distributed_array(segEntry.a, segSubdoms, filenames, dset + "/" + SEGMENTED_OFFSET_NAME, skips);
+        fixupSegBoundaries(segEntry.a, segSubdoms, subdoms);
+        h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(), "Segment Values: %jt".format(segEntry.a));
+
+        var rtnMap: map(string, string) = new map(string, string);
+
+        select dataclass {
+            when C_HDF5.H5T_INTEGER {
+                var (v, idx) = maxloc reduce zip(validFiles, validFiles.domain);
+                if isSigned {
+                    // Load the values
+                    var valEntry = new shared SymEntry(len, int);
+                    read_files_into_distributed_array(valEntry.a, subdoms, filenames, dset + "/" + SEGMENTED_VALUE_NAME, skips);
+                    h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(), "Values Entry Values: %jt".format(valEntry.a));
+
+                    if isBoolDataset(filenames[idx], dset + "/" + SEGMENTED_VALUE_NAME) {
+                        var entryBool = new shared SymEntry(len, bool);
+                        entryBool.a = valEntry.a:bool;
+                        var saEntry = assembleSegArrayFromParts(segEntry, entryBool, st, bool);
+                    
+                        saEntry.fillReturnMap(rtnMap, st);
+                    } else {
+                        // Not a boolean dataset, so add original SymEntry to SymTable
+                        var saEntry = assembleSegArrayFromParts(segEntry, valEntry, st, int);
+                    
+                        saEntry.fillReturnMap(rtnMap, st);
+                    }
+                } else {
+                    // Load the values
+                    var valEntry = new shared SymEntry(len, uint);
+                    read_files_into_distributed_array(valEntry.a, subdoms, filenames, dset + "/" + SEGMENTED_VALUE_NAME, skips);
+
+                    if isBoolDataset(filenames[idx], dset + "/" + SEGMENTED_VALUE_NAME) {
+                        var entryBool = new shared SymEntry(len, bool);
+                        entryBool.a = valEntry.a:bool;
+                        var saEntry = assembleSegArrayFromParts(segEntry, entryBool, st, bool);
+                    
+                        saEntry.fillReturnMap(rtnMap, st);
+                    } else {
+                        // Not a boolean dataset, so add original SymEntry to SymTable
+                        var saEntry = assembleSegArrayFromParts(segEntry, valEntry, st, uint);
```

Comment:
*Duplicate of previous comment:*
If we create the distributed array as suggested and then the SegArray, we do not need `assembleSegArrayFromParts` because we can call the existing `getSegArray(segDist, valDist, st)`.

Reply:
Updated